rfcs/0005-multi-blockchain-support.md

```diff
@@ -0,0 +1,487 @@
+# RFC-0003: Multi-Blockchain Support
+
+<dl>
+  <dt>Author</dt>
+  <dd>Jannis Pohlmann</dd>
+
+  <dt>RFC pull request</dt>
+  <dd><a href="https://github.com/graphprotocol/rfcs/pull/8">https://github.com/graphprotocol/rfcs/pull/8</a></dd>
+
+  <dt>Obsoletes (if applicable)</dt>
+  <dd>-</dd>
+
+  <dt>Date of submission</dt>
+  <dd>2019-12-20</dd>
+
+  <dt>Date of approval</dt>
+  <dd>-</dd>
+
+  <dt>Approved by</dt>
+  <dd>-</dd>
+</dl>
+
+## Contents
+
+<!-- toc -->
+
+## Summary
+
+Multi-blockchain support allows subgraphs to index data from blockchains other
+than Ethereum.
+
+## Goals & Motivation
+
+The main objective of multi-blockchain support is to add support for subgraphs
+to index data from a growing number of blockchains. At the time of writing this
+RFC, only Ethereum and blockchains compatible with the Ethereum JSON-RPC API are
+supported.
+
+This feature unlocks the same use cases on other blockchains that The Graph has
+made possible on Ethereum, including indexing historical data, interacting with
+smart contracts (if supported by the respective blockchain), handling reorgs (if
+applicable) and triggering off of blockchain-specific features like events or
+transactions.
+
+## Urgency
+
+Since this is a big feature and one that requires a substantial refactoring of
+the existing codebase, and _may_ introduce breaking changes, it would be good to
+start working on this as soon as possible.
+
+## Terminology
+
+The feature proposed in this RFC is referred to as **multi-blockchain support**.
+Other terms that are introduced in this RFC or are relevant when talking about
+it are listed below.
+
+- **Network** - a decentralized technology or similar, such as Ethereum,
+  Substrate, or IPFS.
+- **Blockchain** - a blockchain technology, such as Ethereum or Substrate.
+- **Chain** - a specific instance of a _blockchain_, such as Ethereum mainnet.
+- **Chain subgraph** - a built-in subgraph that indexes data intrinsic to a
```

Comment:
We are trying to use the vocabulary that `Network` is a certain blockchain technology, and `Chain` the concrete incarnation of that technology. Calling it a 'Network Subgraph' would muddy this distinction, since the subgraph is specific to a chain. IOW, with what we have, Ethereum is the network, and `kovan` a chain, and the subgraph would be for `kovan` data. Though all this is moot for the time being since such a subgraph is an optional addition that we'll work on after we have the basics of multiblockchain right.